### DISCUSSION TITLE
How to support Eurotronic Comet Zigbee, instead of Eurotronic Spirit Zigbee?

### DISCUSSION THREAD

**Discussion by Xenon69** [Koenkk/zigbee2mqtt#19363] [Devices]
I have some Comet zigbee from Eurotronic, partially recognized as Spirit Zigbee

for example: the valve position is showh at 39% instead 100% when fully open.

I do not want ot suppress Spirit Zigbee support, as I already have several working.

What can be done to add this new model?

Thank you!!

**Comment by Xitee1** [comment]
I would love to see support for the new comet ones too. The spirits are no longer available (at least in Germany). I've just bought a spirit off of eBay (which was really hard to find) but for the other ones I might prefer the newer ones if they also support direct valve control.

**Reply by Xitee1** [reply]
I've now also bought a Comet from Amazon and indeed it gets recognised as Spirit and shows the "PI heating demand" at 39% when fully open. But so far I haven't had any problems with heating. It's even a bit better than the Spirit because the Spirit I have resets to auto mode each day. So far the Comet stayed in manual for 3 days.
One weird thing I noticed is that now the actual current valve position displayed on the valve is the direct value that is set in z2m instead of a calculated % value. And if I set it to like 101 it only shows "On" on the display and if I open it further, the motor still runs.

**Comment by Xenon69** [comment]
What I tryed so far:
- pair a Comet 
=> recognized as a Spirit, with the model name set to SPZB0001
- add a file named COZB0001.js (with minimal content, see below) next to configuration.yaml, added section'external_converters' and restarted Zigbee2mqtt
=> still recognized as a Spirit, with the model name set to SPZB0001

=> How tells Zigbee2mqtt that now there are 2 eurotronic thermostats, one 'SPZB0001', one 'COZB0001'?

I think that it is needed to update the source file zigbee-herdsman-converters/src/devices/eurotronic.ts to prevent Comet devices recognized as Spirit devices...

NB : 'SPZB0001' and 'COZB0001' are identifiers found also in the manufacturer documentation, along with the description of all fields available (setpoint, room temperature, valve position, etc.)

COZB0001.js:

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['COZB0001'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'COZB0001', // Vendor model number, look on the device for a model number
    vendor: 'Eurotronic', // Vendor of the device (only used for documentation and startup logging)
    description: 'Comet thermostat', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [], // We will add this later
    toZigbee: [], // Should be empty, unless device can be controlled (e.g. lights, switches).
    exposes: [e.battery(), e.climate().withSetpoint('occupied_heating_setpoint', 5, 28.5, 0.5)], // Defines what this device exposes, used for e.g. Home Assistant discovery and in the frontend
};

module.exports = definition;
```

**Reply by mabl** [reply]
I'm very new to Zigbee and this code but are we sure that this device actually returns a proper `modelId`? Using the `zigbee2mqtt`  dev console on a Comet ZigBee returns `Read result of 'genBasic': {"modelId":"SPZB0001"}` for me. That would indicate that the device does not return `COZB0001` as indicated on some online documentation. So if I get this right, there is no way of fixing, this apart from a firmware upgrade, no?

**Reply by mabl** [reply]
The documentation also says that the Product Code attribute (0x000a) in the basic cluster is equal to 1002 for the comet. Unfortunately it seems that the attribute `0xa==10` is not defined in herdsman, see https://github.com/Koenkk/zigbee-herdsman/blob/master/src/zcl/definition/cluster.ts#L46. Even if it were defined, I don't yet understand if it could be used to filter the device match for.
